test/unit/model/consent.test.ts

```diff
@@ -0,0 +1,254 @@
+ --------------
+ ******/
+
+import Knex from 'knex'
+import Config from '../../../config/knexfile'
+import ConsentDB, { Consent } from '../../../src/model/consent'
+
+/*
+ * Mock Consent Resources
+ */
+const partialConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123'
+}
+
+const completeConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  credentialId: '123',
+  credentialType: 'FIDO',
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs',
+  credentialPayload: 'dwuduwd&e2idjoj0w'
+}
+
+/*
+ * Consent Resource Model Unit Tests
+ */
+describe('src/model/consent', (): void => {
+  let Db: Knex
+  let consentDB: ConsentDB
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+    await Db.raw('PRAGMA foreign_keys = ON')
+
+    consentDB = new ConsentDB(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  describe('register', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+    })
+
+    it('adds consent with partial info to the database', async (): Promise<void> => {
+      // Action
+      const numInserted: number = await consentDB.register(partialConsent)
+
+      expect(numInserted).toEqual(1)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: partialConsent.id
+        })
+
+      expect(consents[0]).toEqual({
+        id: '1234',
+        initiatorId: 'pisp-2342-2233',
+        participantId: 'dfsp-3333-2123',
+        createdAt: expect.anything(),
+        credentialId: null,
+        credentialType: null,
+        credentialStatus: null,
+        credentialPayload: null,
+        credentialChallenge: null
+      })
+    })
+
+    it('returns an error on adding a consent with existing consentId', async (): Promise<void> => {
+      // Action
+      await consentDB.register(partialConsent)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: partialConsent.id
+        })
+
+      // Consent has been added
+      expect(consents[0]).toEqual({
+        id: '1234',
+        initiatorId: 'pisp-2342-2233',
+        participantId: 'dfsp-3333-2123',
+        createdAt: expect.anything(),
+        credentialId: null,
+        credentialType: null,
+        credentialStatus: null,
+        credentialPayload: null,
+        credentialChallenge: null
+      })
+
+      // Fail primary key constraint
+      await expect(consentDB.register(partialConsent))
+        .rejects.toThrowError()
+    })
+  })
+
+  describe('updateCredentials', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
```

Comment:
There should be a way for SQLite should auto-clear the data automatically. But have to double check....

Reply:
We can destroy the DB connection and start up a new connection for each test but that would take more time than just clearing the tables. I haven't been able to find an other information on this.